src/main/java/reiff/fallingsand/Sand.java

```diff
@@ -33,16 +46,72 @@ public void put(int x, int y) {
     }
 
     public void fall() {
-        for (int y = 1; y >= 0; y--) {
-            for (int x = 0; x < 3; x++) {
-                if (field[y][x] == 1 && field[y + 1][x] == 0) {
-                    // Move sand down
-                    field[y][x] = 0;
-                    field[y + 1][x] = 1;
+        // moves all sand down one square
+        for (int y = field.length - 2; y >= 0; y--) {
+            for (int x = 0; x < field[y].length; x++) {
+                if (field[y][x] == 1) {
+                    if (field[y + 1][x] == 0) {
+                        field[y][x] = 0;
+                        field[y + 1][x] = 1;
+                        continue;
+                    }
+
+                    boolean rightFirst = random.nextBoolean();
+                    int direction1 = rightFirst ? +1 : -1;
+                    int direction2 = rightFirst ? -1 : +1;
+
+                    if (field[y + 1][x + direction1] == 0) {
+                        field[y][x] = 0;
+                        field[y + 1][x + direction1] = 1;
+                    } else if (field[y + 1][x + direction2] == 0) {
+                        field[y][x] = 0;
+                        field[y + 1][x + direction2] = 1;
+                    }
                 }
             }
         }
     }
 
+    public void randomSand(int n) {
+        int freeSpaces = countFreeSpaces();
```

Comment:
This is called once in the beginning of the program, so it's always going to be large

Reply:
I am not sure I understand what you mean by this.

Reply:
There will always be enough free space